scona/scripts/visualisation_commands.py

```diff
@@ -54,22 +54,47 @@ def rescale(fname, suff='png'):
     # And you're done!
 
 
-def view_corr_mat(corr_mat_file,
+def view_corr_mat(corr_mat,
                   output_name,
                   cmap_name='RdBu_r',
                   cost=None,
                   bin=False):
-    ''' This is a visualisation tool for correlation matrices'''
+    '''
+    This is a visualisation tool for correlation matrices
+
+    Parameters
+    ----------
+    corr_mat : :class:`pandas.DataFrame` or :class:`str`
+        corr_mat could be a DataFrame object to represent a correlation matrix
+        or a string object - Path to the File, containing the matrix.
+    output_name : :class:`str`
+        the name of the file you want to save your visualization
+        of correlation matrix to in.
+    cmap_name : string or Colormap, optional
+        A Colormap instance or registered colormap name.
+        The colormap maps scalar data to colors. It is ignored for RGB(A) data.
+        Defaults to 'RdBu_r'.
+
+    Returns
+    -------
+        The visualization of the correlation matrix is saved in the file.
 
-    # Read in the data
-    M = np.loadtxt(corr_mat_file)
+    '''
 
     # If cost is given then roughly threshold at that cost.
     # NOTE - this is not actually the EXACT network that you're analysing
     # because it doesn't include the minimum spanning tree. But it will give
     # you a good sense of the network structure.
     # #GoodEnough ;)
 
+    if isinstance(corr_mat, str):
+        M = np.loadtxt(corr_mat)                  # Read in the data
```

Comment:
Feels like we should add a test for this, what do you think @wingedRuslan?

If that feels a bit too much then we can just open an issue noting that we should check the dimensions of the file when we load it in. Up to you and @Islast ✨

Reply:
@KirstieJane Can you please elaborate on that?
Would you like to check the file size before loading it?

Reply:
Sorry folks! Massively unclear on my part.

The test that @wingedruslan has added further down to check that the data is square is what I was thinking of, not anything about reading in the data properly 😬